packages/grid/x-data-grid/src/components/cell/GridActionsCell.tsx

```diff
@@ -16,76 +17,198 @@ interface TouchRippleActions {
   const buttonRef = React.useRef<HTMLButtonElement>(null);
-  const touchRippleRefs = React.useRef<Record<number, TouchRippleActions | null>>({});
+  const ignoreCallToFocus = React.useRef(false);
+  const touchRippleRefs = React.useRef<Record<string, TouchRippleActions | null>>({});
   const menuId = useId();
   const buttonId = useId();
   const rootProps = useGridRootProps();
-  const { colDef, id, api, hasFocus, position = 'bottom-end' } = props; // TODO apply the rest to the root element
 
   React.useLayoutEffect(() => {
     if (!hasFocus) {
       Object.entries(touchRippleRefs.current).forEach(([index, ref]) => {
         ref?.stop({}, () => {
-          delete touchRippleRefs.current[index as any];
+          delete touchRippleRefs.current[index];
         });
       });
     }
   }, [hasFocus]);
 
+  React.useEffect(() => {
+    if (focusedButtonIndex >= 0) {
+      const child = rootRef.current?.children[focusedButtonIndex] as HTMLElement;
+      child.focus();
+    }
+  }, [focusedButtonIndex]);
+
+  React.useEffect(() => {
+    if (!hasFocus) {
+      setFocusedButtonIndex(-1);
+      ignoreCallToFocus.current = false;
+    }
+  }, [hasFocus]);
+
+  React.useImperativeHandle(
+    focusElementRef,
+    () => ({
+      focus() {
+        // If ignoreCallToFocus is true, then one of the buttons was clicked and the focus is already set
+        if (!ignoreCallToFocus.current) {
+          setFocusedButtonIndex(0);
+        }
+      },
+    }),
+    [],
+  );
+
   if (!hasActions(colDef)) {
     throw new Error('MUI: Missing the `getActions` property in the `GridColDef`.');
   }
 
-  const showMenu = () => setOpen(true);
-
-  const hideMenu = () => setOpen(false);
-
-  const options = colDef.getActions(api.getRowParams(id));
+  const options = colDef.getActions(apiRef.current.getRowParams(id));
   const iconButtons = options.filter((option) => !option.props.showInMenu);
   const menuButtons = options.filter((option) => option.props.showInMenu);
+  const numberOfButtons = iconButtons.length + (menuButtons.length ? 1 : 0);
+
+  const showMenu = () => {
+    setOpen(true);
+    setFocusedButtonIndex(numberOfButtons - 1);
+    ignoreCallToFocus.current = true;
+  };
 
-  const handleTouchRippleRef = (index: number) => (instance: TouchRippleActions | null) => {
-    touchRippleRefs.current[index] = instance;
+  const hideMenu = () => {
+    setOpen(false);
+  };
+
+  const handleTouchRippleRef =
+    (index: string | number) => (instance: TouchRippleActions | null) => {
+      touchRippleRefs.current[index] = instance;
+    };
+
+  const handleButtonClick =
+    (index: number, onClick?: React.MouseEventHandler): React.MouseEventHandler =>
+    (event) => {
+      setFocusedButtonIndex(index);
+      ignoreCallToFocus.current = true;
+
+      if (onClick) {
+        onClick(event);
+      }
+    };
+
+  const handleRootKeyDown = (event: React.KeyboardEvent) => {
+    if (iconButtons.length + menuButtons.length === 1) {
```

Comment:
Should it be
```suggestion
    if (numberOfButtons <= 1) {
```

Because even if `menuButtons` contains lot of options it's still one button so right/left arrow should do nothing